evaluation.py

```diff
@@ -237,7 +237,143 @@ def predict_keypoints(self,generator, location):
+                    left_hip_point = np.array(annotation_keypoints[33], annotation_keypoints[34])
+                    right_hip_point = np.array(annotation_keypoints[36], annotation_keypoints[37])
+                    torso = np.linalg.norm(left_hip_point-right_hip_point)
+                    threshold = PCK_THRESHOLD*torso
+
+                    for i in range(NUM_COCO_KEYPOINTS):
+                        base = i*3
+                        prediction_point = np.array(prediction_keypoints[base], prediction_keypoints[base+1])
+                        annotation_point = np.array(annotation_keypoints[base], annotation_keypoints[base+1])
+                        dist = (np.linalg.norm(prediction_point-annotation_point))
+                        dist_list.append(dist)
+
+            # Each image may have more than one annotation we need to check
+            annotations = int(len(dist_list)/NUM_COCO_KEYPOINTS)
+
+            nose_correct = []
+            left_eye_correct = []
+            right_eye_correct = []
+            left_ear_correct = []
+            right_ear_correct = []
+            left_shoulder_correct = []
+            right_shoulder_correct = []
+            left_elbow_correct = []
+            right_elbow_correct = []
+            left_wrist_correct = []
+            right_wrist_correct = []
+            left_hip_correct = []
+            right_hip_correct = []
+            left_knee_correct = []
+            right_knee_correct = []
+            left_ankle_correct = []
+            right_ankle_correct = []
+
+            # Append True to correct joint list if distance is below threshold for any annotation
+            for j in range(annotations):
+                base = j*NUM_COCO_KEYPOINTS
+                nose_correct.append(dist_list[0+base]<=threshold)
+                left_eye_correct.append(dist_list[1+base]<=threshold)
+                right_eye_correct.append(dist_list[2+base]<=threshold)
+                left_ear_correct.append(dist_list[3+base]<=threshold)
+                right_ear_correct.append(dist_list[4+base]<=threshold)
+                left_shoulder_correct.append(dist_list[5+base]<=threshold)
+                right_shoulder_correct.append(dist_list[6+base]<=threshold)
+                left_elbow_correct.append(dist_list[7+base]<=threshold)
+                right_elbow_correct.append(dist_list[8+base]<=threshold)
+                left_wrist_correct.append(dist_list[9+base]<=threshold)
+                right_wrist_correct.append(dist_list[10+base]<=threshold)
+                left_hip_correct.append(dist_list[11+base]<=threshold)
+                right_hip_correct.append(dist_list[12+base]<=threshold)
+                left_knee_correct.append(dist_list[13+base]<=threshold)
+                right_knee_correct.append(dist_list[14+base]<=threshold)
+                left_ankle_correct.append(dist_list[15+base]<=threshold)
+                right_ankle_correct.append(dist_list[16+base]<=threshold)
```

Comment:
I think this could be made cleaner if you set them as `false` bools in the beginning and then in the for loop you run a `joint_x = joint_x or (condition here)`

Reply:
Oh yeah you know what it would probably make more sense if theyre just left as bools

Reply:
There might even be some nice `numpy` way to do this on the entire vector at once haha but it's all good if we don't

Reply:
Yeah haha I'm sure there's a better way to do it, it just gets finicky with different numbers of annotations for each image